sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -0,0 +1,128 @@
+
+public class CosmosMultiHashTest extends TestSuiteBase {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private String preExistingDatabaseId = CosmosDatabaseForTest.generateId();
+    private CosmosClient client;
+    private CosmosDatabase createdDatabase;
+    private CosmosContainer createdMultiHashContainer;
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosMultiHashTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void before_CosmosItemTest() {
+
+        client = getClientBuilder().buildClient();
+        createdDatabase = createSyncDatabase(client, preExistingDatabaseId);
+        String collectionName = UUID.randomUUID().toString();
+
+        PartitionKeyDefinition partitionKeyDefinition = new PartitionKeyDefinition();
+        partitionKeyDefinition.setKind(PartitionKind.MULTI_HASH);
+        partitionKeyDefinition.setVersion(PartitionKeyDefinitionVersion.V2);
+        ArrayList<String> paths = new ArrayList<>();
+        paths.add("/city");
+        paths.add("/zipcode");
+        partitionKeyDefinition.setPaths(paths);
+
+        CosmosContainerProperties containerProperties = getCollectionDefinition(collectionName, partitionKeyDefinition);
+
+        //MultiHash collection create
+        CosmosContainerResponse containerResponse = createdDatabase.createContainer(containerProperties);
+
+        //MultiHash collection read
+        createdMultiHashContainer = createdDatabase.getContainer(collectionName);
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        logger.info("starting ....");
+        //MultiHash Collection delete
+        createdMultiHashContainer.delete();
+        safeDeleteSyncDatabase(createdDatabase);
+        safeCloseSyncClient(client);
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void itemCRUDTest() throws Exception {
+
+        List<String> pkIds = new ArrayList<>();
+        pkIds.add("Redmond");
+        pkIds.add("98052");
+
+        Object[] pkValue = new Object[2];
+        pkValue[0] = pkIds.get(0);
+        pkValue[1] = pkIds.get(1);
+
+        PartitionKey partitionKey = new PartitionKey(pkValue);
+
+        String documentId = UUID.randomUUID().toString();
+        ObjectNode properties = getDocumentDefinition(documentId, pkIds);
+        createdMultiHashContainer.createItem(properties);
+
+        CosmosItemResponse<ObjectNode> readResponse1 = createdMultiHashContainer.readItem(
+                                                        documentId, partitionKey, ObjectNode.class);
+        validateIdOfItemResponse(documentId, readResponse1);
+        assertThat(readResponse1.getItem().equals(properties));
+    }
+
+    private ObjectNode getDocumentDefinition(String documentId, List<String> pkIds) throws JsonProcessingException {
```

Comment:
as tests are open source they will be used as samples by end user, please don't use the term Document.
We are moving away from Document concept.

```suggestion
    private ObjectNode getItem(String documentId, List<String> pkIds) throws JsonProcessingException {

```